daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -179,7 +179,7 @@ trait ElementBaseGrammarMixin
         prefixLengthType
       )
       schemaDefinitionWhen(
-        detachedElementDecl.alignment != 1,
+        detachedElementDecl.alignment.asInstanceOf[Int] != 1,
```

Comment:
The asInstanceOf call is a setting off my code smell senses, but it looks like its necessary given how AlignmentType is defined using AnyRef.

Reply:
Scala 3 has disjunction and union types which in theory allow cleaner solutions to this sort of thing. Worth learning about in the future. 

For this change set I'm ok with just the cast here.